### DISCUSSION TITLE
inla.cpo for inlabru

### DISCUSSION THREAD

**Discussion by Mouniaz96** [inlabru-org/inlabru#171] [Q&A] [answered]
Dear all,

I have a count data (poisson distribution) and I got from the cpo results some failures and a PIT histogram with a lot of values close to 0 and 1. I read that we could avoid to recompute manually the CPO/PIT values for the observations in violation by using directly the inla.cpo function. However, it seems to work only for inla objects, is there a way to do something similar with inlabru inputs ?
Regards

Mounia

**Comment by finnlindgren** [comment, marked as answer]
The ouptut from `bru()` is an `inla` object with added information, so anything you can do with regular `inla()` output should also work with `bru()` output. What seems to go wrong when you try it?

**Reply by finnlindgren** [reply]
What I can see going wrong is if you then try to apply inlabru methods (like predict) on the output from `inla.cpo`, since that might no longer contain the inlabru specific information.
I need more information.

**Reply by Mouniaz96** [reply]
Hi,
I just tried to apply the inla.cpo function on the output from bru() and I had this message : Error in if (class(result) != "inla") { : the condition has length > 1
Would you like me to send you my code or output ?

**Reply by finnlindgren** [reply]
Ok, that’s a bug in the inla code; it should use `inherits(result, “inla”)` to check the object class. I’ll look into getting that fixed.

**Reply by finnlindgren** [reply]
I believe this is now fixed in the INLA development code, https://github.com/hrue/r-inla/commit/c6134f5396598a34787b4c90cd7275b2268b0ad2
It should be available in the INLA package from the next build (the latest was yesterday, 22.11.08).

**Reply by Mouniaz96** [reply]
Ok thank you. So should I reinstall INLA (the one I use is 22.07.23) ?

**Reply by finnlindgren** [reply]
Not yet; for this feature you need to wait until the next build, i.e. when the version number increases beyond 22.11.08.

As a temporary workaround, you can alter the class information before the `inla.cpo` call and then restore it afterwards:
```
old_class <- class(result)
class(result) <- "inla"
result <- inla.cpo(result)
class(result) <- old_class
```

**Reply by Mouniaz96** [reply]
Ok thank you very much

**Reply by Mouniaz96** [reply]
Hi Finn. Just a question, I did what you suggested in the meantime and I'm still waiting for the results, several hours after, with this message : Compute new CPO/PIT values manually, for 32 cases...
as(<dgCMatrix>, "dgTMatrix") is deprecated since Matrix 1.5-0; do as(., "TsparseMatrix") instead.
 Is it normal knowing that there are only 32 failures out of 7440 ?

**Reply by finnlindgren** [reply]
The Matrix issue has been fixed in recent INLA and inlabru builds, so upgrading INLA should solve it (and it still works with your current version; you get the warnings but it’s still doing what it’s meant to do).
